PyRoute/TradeCalculation.py

```diff
@@ -756,6 +783,25 @@ def route_weight(self, star, target):
         weight -= star.importance + target.importance
         return weight
 
+    def calculate_components(self):
+        bitz = nx.connected_components(self.galaxy.stars)
+        raw_components = []
+        total_stars = 0
+
+        for bit in bitz:
+            if 1 == len(bit):
```

Comment:
Interesting bonus challenge: I would love to have this hard coded "1" be configurable. At minimum I would like to set this to 3.

Reply:
How would you want to make it configurable?  Class-level variable?  (something like `self.min_component_size`)

Reply:
@tjoneslo , I went with a class-level variable - it seemed simplest.